### PR TITLE
fix(button): show correct activated state for ios

### LINKED ISSUE #22468: bug: hover state covers activated state on button
# Bug Report

**Ionic version:**
[x] **5.x**

**Describe the Feature Request**
The buttons on iOS should have a better UX. More life when buttons pressing.

**Describe Preferred Solution**
Just a better indication for user input. The user should not ask, "Did I press the button or not?"
Of course, a button action mostly results to screen change. (change value, page, etc.)
But I think, it's not the worst idea to breathe more life into the button. Optional or not.

**Describe Alternatives**
Option to enable ripple effect for iOS buttons too. But a fading highlight will also better than a static element. Something you can see, when your finger is not on screen anymore. A short animation time like ripple-effect. Immediately on / after pressing.

**Related Code**
https://ionicframework.com/docs/api/button

**Additional Context**

| md | ios |
| :-: | :-: |
| md-button | ios-button |
|wow-reaction  | disappointed-reaction |

On iOS the only click indicator is from my gif recorder. 😄

### PR DESCRIPTION
**Pull request by brandyscarney**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- The activated state does not show up on iOS while hovering
- Outline buttons in a toolbar aren't showing activated state at all

Issue URL: fixes #22468

## What is the new behavior?

- Display activated states for buttons when pressing while hovering
- Show the activated state for outline buttons in a toolbar (updated to match what an outline button with color was already doing) 
- While fixing the above there were some issues that popped up, resolved all found issues 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

The following tests should show no differences:

- button/basic
- button/clear
- button/outline

The following tests will show fixes where activated was not being applied at all for outline buttons, regardless of hover state. I made edits to these tests on my branch in order to check more use cases, so these diffs won't show up in screenshot comparisons:

| test | main | branch | 
| ---| ---| ---|
| toolbar/basic | toolbar - basic 1 | toolbar - basic 2 | 
| toolbar/colors | toolbar - colors 2 | toolbar - colors 1 |
| themes/default | themes - default | themes - default |
| themes/oceanic | themes - oceanic | themes - oceanic |
| themes/vibrant | themes - vibrant |  themes - vibrant |
| themes/dark |  themes - dark | themes - dark |

### CODE REVIEW

File: core/src/components/back-button/back-button.ios.scss [hunk 1]
```diff
@@ -38,11 +38,3 @@
...
   }
 }
-
-
-// Back Button Color: Focused
-// --------------------------------------------------
-
-:host(.ion-color.ion-focused) .button-native::after {
-  background: #{current-color(base)};
-}
```
Comment (amandaejohnston) [inline core/src/components/back-button/back-button.ios.scss:48]: Why was this removed?

File: core/src/components/toolbar/test/colors/index.html [hunk 2]
```diff
@@ -36,147 +34,140 @@
...
             <ion-buttons slot="primary">
-              <ion-button fill="clear">Clear</ion-button>
               <ion-button>
                 <ion-icon slot="icon-only" ios="ellipsis-horizontal" md="ellipsis-vertical"></ion-icon>
               </ion-button>
             </ion-buttons>
             <ion-title>Custom</ion-title>
           </ion-toolbar>
+
+          <ion-toolbar class="custom-toolbar">
```
Comment (amandaejohnston) [inline core/src/components/toolbar/test/colors/index.html:139]: Some of the screenshots are getting cut off down here, examples:
- `core/src/components/toolbar/test/colors/toolbar.e2e.ts-snapshots/toolbar-colors-md-ltr-Mobile-Chrome-linux.png`
- `core/src/components/toolbar/test/colors/toolbar.e2e.ts-snapshots/toolbar-colors-md-ltr-Mobile-Firefox-linux.png`
- `core/src/components/toolbar/test/colors/toolbar.e2e.ts-snapshots/toolbar-colors-md-ltr-Mobile-Safari-linux.png`

Comment (brandyscarney) [reply core/src/components/toolbar/test/colors/index.html:139]: Looking into fixing this!

Comment (brandyscarney) [reply core/src/components/toolbar/test/colors/index.html:139]: Should be fixed now.

**Review by amandaejohnston — APPROVED** [review]
